swap/swap.go

```diff
@@ -577,27 +599,39 @@ func (s *Swap) getContractOwner(ctx context.Context, address common.Address) (co
 	return contr.Issuer(nil)
 }
 
-func promptInitialDepositAmount() (uint64, error) {
-	// need to prompt user for initial deposit amount
-	// if 0, can not cash in cheques
+// promptDepositAmount blocks and asks the user how much ERC20 he wants to deposit
+func (s *Swap) promptDepositAmount() (*big.Int, error) {
+	// retrieve available balance
+	availableBalance, err := s.AvailableBalance()
+	if err != nil {
+		return nil, err
+	}
+	balance, err := s.contract.BalanceAtTokenContract(nil, s.owner.address)
+	if err != nil {
+		return nil, err
+	}
+	// log available balance and ERC20 balance
+	swapLog.Info("Balance information", "chequebook available balance", availableBalance, "ERC20 balance", balance)
+	promptMessage := fmt.Sprintf("Please provide the amount in HONEY which will deposited to your chequebook (0 for skipping deposit): ")
+	// need to prompt user for deposit amount
 	prompter := console.Stdin
-
 	// ask user for input
-	input, err := prompter.PromptInput("Please provide the amount in Wei which will deposited to your chequebook upon deployment: ")
+	input, err := prompter.PromptInput(promptMessage)
+	var zero big.Int
 	if err != nil {
-		return 0, err
+		return &zero, err
 	}
 	// check input
 	val, err := strconv.ParseInt(input, 10, 64)
 	if err != nil {
 		// maybe we should provide a fallback here? A bad input results in stopping the boot
```

Comment:
a possibility is booting without depositing in case there is an invalid input. 

not sure if this is the best option for UX

Reply:
I would ideally like to see something similar to the password prompt when you boot up Swarm (or Geth, it is a legacy from there...). 
However, this involves a bit more coding and in the meantime I think just stopping the boot is fine.

Reply:
agreed